### DISCUSSION TITLE
[SDK-813] Remix: Provide Non-Root Implementation

### DISCUSSION THREAD

**Discussion by parisholley** [clerk/javascript#1963] [Feature Requests]
### Preliminary Checks

- [X] I have reviewed https://clerk.com/docs for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/clerkinc/javascript/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to Clerk. Please ask questions in our Discord community: https://clerk.com/discord.

### Description

The current documentation encourages the user to create a root loader and wrap the main App with `ClerkApp` however this is not an efficient implementation for a few reasons:

1. There is no way to avoid making a call to the ClerkAPI every request, if some routes are unauthenticated, it is wasted latency.
2. Any use of `getAuth()` in a loader will cause additional requests as there is no request-level cache (eg: storing promise on the remix context, keeping in mind that loaders execute in parallel).

Rather than only offer `ClerkApp` as a solution, encourage the user use `ClerkProvider` directly and provide a `getClerkState()` call that returns valid types for TS.

### Affected Packages

@clerk/remix

<sub>[SDK-813](https://linear.app/clerk/issue/SDK-813/remix-provide-non-root-implementation)</sub>

**Comment by LekoArts** [comment]
Hello!

We're tracking our feature requests now in [feedback.clerk.com](https://feedback.clerk.com). If your suggestion is already in the list of proposed changes, please upvote. Otherwise, please create an entry. Thanks!

Our feedback platform will be the source of truth for feature requests going forward and your input and votes will help us shape the roadmap.

Thanks for using Clerk! 💙

**Comment by AdiRishi** [comment]
Super +1 on this. It would be a huge win given many apps can have a large "logged out" section of routes where there is no need to call the clerk auth endpoint.

**Reply by alopes** [reply]
Any updates on this?

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
